src/formatDate.js

```diff
@@ -51,6 +51,31 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateInArr = date.split(fromFormat[fromFormat.length - 1]);
+  const dateOutArr = [];
+
+  for (const keyto of toFormat) {
+    for (const keyfrom of fromFormat) {
```

Comment:
Loop-in-the-loop is a very resource-intensive decision.  In more complicated tasks, execution may overflow the stack.